emcc.py

```diff
@@ -1265,15 +1265,14 @@ def is_supported_link_flag(f):
       if shared.Settings.USE_PTHREADS:
         shared.Settings.FETCH_WORKER_FILE = unsuffixed(os.path.basename(target)) + '.fetch.js'
 
-    if shared.Settings.FETCH:
-      # In asm.js+pthreads we can use a fetch worker, which is made from the main
-      # asm.js code. That lets us do sync operations by blocking on the worker etc.
-      # In the wasm backend we don't have a fetch worker implemented yet, however,
-      # we can still do basic synchronous fetches in the same places: if we can
-      # block on another thread then we aren't the main thread, and if we aren't
-      # the main thread then synchronous xhrs are legitimate.
-      if shared.Settings.USE_PTHREADS and not shared.Settings.WASM_BACKEND:
-        shared.Settings.USE_FETCH_WORKER = 1
+    # In asm.js+pthreads we can use a fetch worker, which is made from the main
+    # asm.js code. That lets us do sync operations by blocking on the worker etc.
+    # In the wasm backend we don't have a fetch worker implemented yet, however,
+    # we can still do basic synchronous fetches in the same places: if we can
+    # block on another thread then we aren't the main thread, and if we aren't
+    # the main thread then synchronous xhrs are legitimate.
+    if shared.Settings.FETCH and shared.Settings.WASM_BACKEND:
```

Comment:
I think this also needs to take into account `USE_PTHREADS`. Later down we check just `USE_FETCH_WORKER`, so if that is defaulted to 1, we'll try to create one in non-pthreads builds.

Reply:
Good point.